browser/brave_ads/application_state/notification_helper/notification_helper_impl_win.cc

```diff
@@ -59,12 +65,64 @@ enum FocusAssistResult {
   ALARMS_ONLY = 2
 };
 
-///////////////////////////////////////////////////////////////////////////////
+// Templated wrapper for ABI::Windows::Foundation::GetActivationFactory()
+template <unsigned int size>
+HRESULT CreateActivationFactory(wchar_t const (&class_name)[size],
+                                const IID& iid,
+                                void** factory) {
+  auto ref_class_name =
+      base::win::ScopedHString::Create(std::wstring_view(class_name, size - 1));
```

Comment:
this potentially should be something like

```cxx
base::as_string_view(base::span(class_name).first<size-1>())
```

Reply:
Fixed

Reply:
Discussed in DM with @cdesouza-chromium  and decided to revert change with span:
- the current code is safe because `class_name` is fixed size null terminated array
- chromium uses the same code

cc @cdesouza-chromium, please correct me if I'm wrong